com.unity.multiplayer.mlapi/Runtime/Core/RPCMethods/NetworkedBehaviour.PerformanceClientRpc.cs

```diff
@@ -81,55 +81,89 @@ public void InvokeClientRpcOnEveryoneExcept(string methodName, ulong clientIdToI
 		{
-			SendClientRPCPerformance(HashMethod(method.Method), stream, clientIdToIgnore, channel, security);
+            List<ulong> clientIds = InternalMessageSender.GetAllClientIdsExcluding(clientIdToIgnore);
+            if(clientIds.Count > 0)
+            {
+			    InvokeClientRPCPerformance(HashMethod(method.Method),clientIds, stream, channel, security);
+            }
 		}
 
 		public void InvokeClientRpcPerformance(string methodName, List<ulong> clientIds, Stream stream, string channel = null, SecuritySendFlags security = SecuritySendFlags.None)
 		{
-			SendClientRPCPerformance(HashMethodName(methodName), clientIds, stream, channel, security);
+			InvokeClientRPCPerformance(HashMethodName(methodName), clientIds, stream, channel, security);
 		}
 
 		public void InvokeClientRpcOnClientPerformance(string methodName, ulong clientId, Stream stream, string channel = null, SecuritySendFlags security = SecuritySendFlags.None)
 		{
-			SendClientRPCPerformance(HashMethodName(methodName), clientId, stream, channel, security);
+			InvokeClientRPCPerformance(HashMethodName(methodName), new List<ulong>(){clientId }, stream, channel, security);
 		}
 
-		public void InvokeClientRpcOnOwnerPerformance(string methodName, Stream stream, string channel = null, SecuritySendFlags security = SecuritySendFlags.None)
+		public void InvokeClientRpcOnEveryonePerformance(string methodName, Stream stream, string channel = null, SecuritySendFlags security = SecuritySendFlags.None)
 		{
-			SendClientRPCPerformance(HashMethodName(methodName), OwnerClientId, stream, channel, security);
+			InvokeClientRPCPerformance(HashMethodName(methodName), InternalMessageSender.GetAllClientIds(), stream, channel, security);
 		}
 
-		public void InvokeClientRpcOnEveryonePerformance(string methodName, Stream stream, string channel = null, SecuritySendFlags security = SecuritySendFlags.None)
+		public void InvokeClientRpcOnEveryoneExceptPerformance(string methodName, ulong clientIdToIgnore, Stream stream, string channel = null, SecuritySendFlags security = SecuritySendFlags.None)
 		{
-			SendClientRPCPerformance(HashMethodName(methodName), null, stream, channel, security);
+			InvokeClientRPCPerformance(HashMethodName(methodName),InternalMessageSender.GetAllClientIdsExcluding(clientIdToIgnore), stream,  channel, security);
 		}
 
-		public void InvokeClientRpcOnEveryoneExceptPerformance(string methodName, ulong clientIdToIgnore, Stream stream, string channel = null, SecuritySendFlags security = SecuritySendFlags.None)
+        internal void InvokeClientRPCPerformance(ulong methodHash, List<ulong> clientIds, Stream stream, string channel = null, SecuritySendFlags security = SecuritySendFlags.None)
+        {
+
+            if (IsHost)
+            {
+                SendClientRPCPerformance(methodHash,clientIds, stream,  channel, security);
+            }
+            if(clientIds != null)
+            {
+                //PerformanceQueueItem QueueItem = new PerformanceQueueItem();
```

Comment:
Do we mean to leave in this commented code?  if so, worth a comment saying what it's for / when we'd want to resurrect it.  Else delete

Reply:
Also, if we don't use the clientIds, should we revamp InvokeClientRPCPerformance to not receive that parameter?

Reply:
That is a good point, I believe that is code that will become retired with the newer standard rpc api taking its place.
So, rule of thumb currently is "if it is convenient or performant rpc code then it might no longer exist in the very near future".
However, you are right in catching the commented code...that is bad form on my part indeed!

Reply:
Well again I don't really mind hunks of commented out code so long as it has some kind of note like "keeping until conv/perf RPC system eliminated" so that someday someone else can know if it should be kept around else it will live forever :)